client/src/components/BlockLinkField/tests/BlockLinkField-test.js

```diff
@@ -1,305 +1,111 @@
-          {...props}
-        />
-      );
-
-      expect(wrapper.find('.block-link-field__actions')).toHaveLength(1);
-      expect(wrapper.find(mockedBlockLinkFieldActionsComponent)).toHaveLength(1);
-    });
-  });
-
-  describe('handleSubmitModal()', () => {
-    it('Handles the insert link modal form submission', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
-
-      wrapper.instance().handleCloseModal = mockHandleCloseModal;
-      wrapper.instance().registerChange = mockRegisterChange;
-
-      const mockData = {
-        SecurityID: 1,
-        action_insert: false,
-        content: 'test-mock-data'
-      };
-
-      wrapper.setState({ value: { content: '' } });
-      wrapper.instance().handleSubmitModal(mockData);
-
-      expect(wrapper.state('value')).toEqual({ content: 'test-mock-data' });
-      expect(mockRegisterChange).toHaveBeenCalled();
-      expect(mockHandleCloseModal).toHaveBeenCalled();
-    });
-  });
-
-  describe('handleCloseModal()', () => {
-    it('triggers the insert link modal to close', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
-      wrapper.setState({ modalOpen: true });
-      wrapper.instance().handleCloseModal();
-
-      expect(wrapper.state('modalOpen')).toEqual(false);
-    });
-  });
-
-  describe('handleKeyUp()', () => {
-    it('triggers the handleClick function', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
-
-      wrapper.instance().handleClick = mockHandleClick;
-
-      const mockEvent = {
-        keyCode: 13
-      };
-
-      wrapper.instance().handleKeyUp(mockEvent);
-
-      expect(mockHandleClick).toHaveBeenCalledWith(mockEvent);
-    });
-  });
+    value: '{"PageID":1}',
+    toggle: () => {},
+    BlockLinkFieldActionsComponent: () => <div className="test-block-link-field-actions" />,
+    // using a change handler to trigger the insert handler
+    InsertModalLinkComponent: ({onInsert}) => <input className="test-insert-modal-link" onChange={(evt) => onInsert(evt)} />,
+    ...obj
+  }
+}
+
+test('BlockLinkField should render the class icon, the link, and the actions', () => {
+  const { container } = render(<BlockLinkField {...makeProps()}/>);
+  expect(container.querySelector('.block-link-field__icon')).not.toBeNull()
+  expect(container.querySelector('.block-link-field__content')).not.toBeNull()
+  expect(container.querySelector('.block-link-field__actions')).not.toBeNull()
+});
 
-  describe('handleClick()', () => {
-    it('triggers the insert link modal to open', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
+test('BlockLinkField should render no actions if BlockLinkActionsComponent is not given', () => {
+  const { container } = render(
+    <BlockLinkField {...makeProps({
+      BlockLinkFieldActionsComponent: null
```

Comment:
Doesn't matter, logic in BlocklinkField is 
```
    if (!BlockLinkFieldActionsComponent) {
      return null;
    }
```

Reply:
Should the test really be dependent on implementation logic? Is it meant to be testing "if there is no prop passed in" or "if the value of the internal variable is falsy"?
In any case given that this is the same as the test used to be I'll let this go to avoid ping pong.